### PR TITLE
blog: GSOC 2024 announcement post

### PR DESCRIPTION
**Pull request by erickzhao**
Hey hey hey @electron/summer-of-code

### CODE REVIEW

File: .github/workflows/semantic.yml [hunk 1]
```diff
@@ -24,3 +24,16 @@ jobs:
...
+            docs
+            style
+            refactor
+            perf
+            test
+            build
+            ci
+            chore
+            revert
+            blog
```
Comment (erickzhao) [inline .github/workflows/semantic.yml:39]: I added `blog` as a valid semantic commit prefix. All others are default.

File: blog/2024-summer-of-code.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+---
+title: Google Summer of Code 2024
+date: 2024-11-23T00:00:00.000Z
```
Comment (dsanders11) [inline blog/2024-summer-of-code.md:3]:
```suggestion
date: 2024-02-23T00:00:00.000Z
```

**Review by VerteDinde — APPROVED** [review]

File: blog/2024-summer-of-code.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+As a Summer of Code contributor, you would be collaborating with some of Electron’s core contributors
+on one of many projects under the [github.com/electron](http://github.com/electron) umbrella.
+
+## Before applying
+
+If you aren’t very familiar with Electron, we would recommend you start by reading the
+[documentation](https://electronjs.org/docs/latest) and trying out examples in [Electron Fiddle](https://electronjs.org/fiddle).
+
+To learn more about Electron app distribution, you can also play around with
+[Electron Forge](https://github.com/electron/forge/blob/main/LICENSE) by creating a sample application:
```
Comment (BlackHole1) [inline blog/2024-summer-of-code.md:48]:
```suggestion
[Electron Forge](https://github.com/electron/forge/blob/main/README.md) by creating a sample application:
```

File: blog/2024-summer-of-code.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+Our 2022 Google Summer of Code intern, [@aryanshridhar](https://github.com/aryanshridhar),
+did an amazing job! If you want to see what Aryan worked on during his summer with Electron,
+you can read his report in the [2022 GSoC program archives](https://summerofcode.withgoogle.com/archive/2022/organizations/electron).
+
+:::
+
+## Questions?
+
+If you have questions we didn’t address in the blog post or inquiries for your proposal draft,
+please send us an email at [summer-of-code@electronjs.org](mailto:summer-of-code@electronjs.org)!
```
Comment (BlackHole1) [inline blog/2024-summer-of-code.md:103]:
```suggestion
please send us an email at [summer-of-code@electronjs.org](mailto:summer-of-code@electronjs.org) or check [GSoC FAQ](https://developers.google.com/open-source/gsoc/faq)!
```

**Review by BlackHole1 — DISMISSED** [review]

**Review by dsanders11 — APPROVED** [review]

**Pull request merged**